xbmc/video/Teletext.cpp

```diff
@@ -2255,7 +2262,11 @@ void CTeletextDecoder::RenderCharIntern(TextRenderInfo_t* RenderInfo, int Char,
 
   /* render char */
   sbitbuffer = m_sBit->buffer;
-  unsigned char localbuffer[1000]; // should be enough to store one character-bitmap...
+
+  std::vector<unsigned char> localbuffer;
```

Comment:
I don't know how this part works (teletext) but I think it's very inefficient to create/destroy a vector to render just one character. I suspect that this method will be called multiple times in a row.

It's much better to just increase the buffer size to 2000, 3000, 5000, 10000 or whatever is needed and keep it on the stack. How many pixels does a teletext character have?

The rest of the changes looks fine.